gcalbot/gcalbot/db.go

```diff
@@ -491,3 +492,100 @@ func (d *DB) GetInviteAndAccountByUserMessage(keybaseUsername string, messageID
 		return nil, nil, err
 	}
 }
+
+// Daily Schedule Subscription
+func (d *DB) InsertDailyScheduleSubscription(account *Account, subscription DailyScheduleSubscription) error {
+	notificationDuration := GetMinutesFromDuration(subscription.NotificationDuration)
+	return d.RunTxn(func(tx *sql.Tx) error {
+		_, err := tx.Exec(`
+			INSERT INTO daily_schedule_subscription
+			(keybase_username, account_nickname, calendar_id, keybase_conv_id, timezone, days_to_send, schedule_to_send, notification_duration)
+			VALUES (?, ?, ?, ?, ?, ?, ?, ?)
+			ON DUPLICATE KEY UPDATE
+			timezone=VALUES(timezone),
+			days_to_send=VALUES(days_to_send),
+			schedule_to_send=VALUES(schedule_to_send),
+			notification_duration=VALUES(notification_duration)
+		`, account.KeybaseUsername, account.AccountNickname, subscription.CalendarID, subscription.KeybaseConvID,
+			subscription.Timezone.String(), subscription.DaysToSend, subscription.ScheduleToSend, notificationDuration)
+		return err
+	})
+}
+
+func (d *DB) GetAggregatedDailyScheduleSubscription(scheduleToSend ScheduleToSendType) (subscriptions []*AggregatedDailyScheduleSubscription, err error) {
+	row, err := d.DB.Query(`
+		SELECT
+			GROUP_CONCAT(calendar_id) as calendar_ids, keybase_conv_id, timezone, days_to_send, schedule_to_send, notification_duration,
+			account.keybase_username, account.account_nickname, access_token, token_type, refresh_token, ROUND(UNIX_TIMESTAMP(expiry))
+		FROM daily_schedule_subscription
+		JOIN account USING(keybase_username, account_nickname)
+		WHERE schedule_to_send = ?
+		GROUP BY keybase_username, account_nickname, keybase_conv_id
+	`, scheduleToSend)
+	if err != nil {
+		return nil, err
+	}
+	defer row.Close()
+	for row.Next() {
+		var pair AggregatedDailyScheduleSubscription
+		var concatCalendarIDs string
+		var timezone string
+		var notificationDuration int
+		var tokenExpiry int64
+		err = row.Scan(&concatCalendarIDs, &pair.KeybaseConvID, &timezone, &pair.DaysToSend, &pair.ScheduleToSend, &notificationDuration,
+			&pair.Account.KeybaseUsername, &pair.Account.AccountNickname, &pair.Account.Token.AccessToken, &pair.Account.Token.TokenType,
+			&pair.Account.Token.RefreshToken, &tokenExpiry)
+		if err != nil {
+			return nil, err
+		}
+		pair.CalendarIDs = strings.Split(concatCalendarIDs, ",")
+		pair.Timezone, err = time.LoadLocation(timezone)
+		if err != nil {
+			return nil, err
```

Comment:
errorf and continue, shouldn't kill the run for everyone

Reply:
I had to add a debugoutput to the db